File: ruby/raindrops/raindrops.rb
class Raindrops
  RAINDROPS  = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }

  # my solution:

  # def self.convert(input)
  #   output = ''
  #   RAINDROPS.each_key do |key|
  #     output << RAINDROPS[key] if (input % key).zero?
  #   end
  #   output.empty? ? input.to_s : output
  # end

  # solution suggested by mentor ajoshguy:

  def self.convert(input)
    output = RAINDROPS.sum("") do |key, value|
      (input % key).zero? ? value : ""
    end
    output.empty? ? input.to_s : output
  end
end

File: javascript/raindrops/raindrops.js
const raindrops = {
  3: 'Pling',
  5: 'Plang',
  7: 'Plong'
};

export default class Raindrops {
  convert(n) {
    let output = '';
    Object.keys(raindrops).forEach(key => {
      if (n % key == 0) output += raindrops[key];
    })
    return output == '' ? n.toString() : output;
  }
}
